sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentField.cs

```diff
@@ -311,5 +326,23 @@ public AddressValue AsAddress()
 
             return ValueAddress;
         }
+
+        private object InternalValue => ValueType switch
+        {
+            DocumentFieldType.Address => AsAddress(),
+            DocumentFieldType.CountryRegion => AsCountryRegion(),
```

Comment:
You mentioned we have seen this pattern in TextAnalytics too. Is this a pattern we should account for by default in our codegen?

i.e. debuggability experience for a type that is essentially being converted from a catch-all type to a strongly typed design in convenience layer will need extra debuggability support.
Is there a swagger check that we can use to avoid this?
If this is a common enough pattern should we consider this in codegen via transform\attributes etc?

Reply:
I actually mentioned we used the [DebuggerTypeProxyAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggertypeproxyattribute?view=net-6.0) in TA, but the scenario there was a bit different ([TA code](https://github.com/Azure/azure-sdk-for-net/blob/6880fd3930bb5cd7531850a0ec88eab109a61c93/sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesResultCollection.cs#L45)). In TA, we wanted to expose a property (`Items`) that was hard to find in the debugger because it was nested inside other elements, so we simply moved it to the root level.

Originally I was thinking about using the `DebuggerTypeProxyAttribute` as well, but after some consideration I realized it was an overkill. The approach I'm taking in this PR uses less code, but I can't say it's a final design.

Reply:
> i.e. debuggability experience for a type that is essentially being converted from a catch-all type to a strongly typed design in convenience layer will need extra debuggability support. Is there a swagger check that we can use to avoid this?

From my understanding the swagger does not have any kind of flag to distinguish this scenario from others. Here's the `DocumentField` definition: [swagger](https://github.com/Azure/azure-rest-api-specs/blob/a69db59e981eae7dc554823100220c81b34225cc/specification/cognitiveservices/data-plane/FormRecognizer/preview/2022-06-30-preview/FormRecognizer.json#L1849)

It only defines all of its value properties (`valueString`, `valueNumber`, `valueTime`, etc) as optional, and we trust the service will send back the one that corresponds to the `type` property.

I've only seen this pattern being used in `DocumentField`, and haven't seen it in other libraries.

Reply:
Thanks Caio